src/pybind/mgr/cephadm/module.py

```diff
@@ -2539,6 +2539,23 @@ def _apply(self, spec: GenericSpec) -> str:
     def apply_tuned_profiles(self, specs: List[TunedProfileSpec]) -> str:
         outs = []
         for spec in specs:
+
+            # ensure all the configured sysctl options are valid
+            if spec.settings is not None:
+                candidate_hosts = spec.placement.filter_matching_hostspecs(self.inventory.all_specs())
+                invalid_options: Dict[str, List[str]] = {}
+                for host in candidate_hosts:
+                    host_sysctl_options = self.cache.get_facts(host).get('sysctl_options', {})
+                    if not host_sysctl_options:
+                        continue
+                    invalid_options[host] = []
+                    for option in spec.settings:
+                        if option not in host_sysctl_options:
+                            invalid_options[host].append(option)
+                if any(e != [] for e in invalid_options.values()):
```

Comment:
```suggestion
                if any(e for e in invalid_options.values()):
```
NIT, since lists evaluate to false if empty or true if not empty anyhow

Reply:
fixed